File: 2149. Rearrange Array Elements by Sign.py
#PYTHON
#* 2149. Rearrange Array Elements by Sign
nums = [3,1,-2,-5,2,-4]
# ANSWER => [3,-2,1,-5,2,-4]
from typing import List
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        positives = [num for num in nums if num > 0]
        negatives = [num for num in nums if num <= 0]
        return [num for group in zip(positives, negatives) for num in group]
    
print(Solution().rearrangeArray(nums))

File: 2149. Rearrange Array Elements by Sign.rs
// 2149. Rearrange Array Elements by Sign
// nums = [3,1,-2,-5,2,-4]
//SOL PARTITION => pos.zip(neg)
impl Solution {
  pub fn rearrange_array(nums: Vec<i32>) -> Vec<i32> {
      let (positives, negatives): (Vec<i32>, Vec<i32>) = nums.into_iter().partition(|&x| x > 0);
      let mut result = vec![];
      
      for (positive, negative) in positives.into_iter().zip(negatives.into_iter()) {
          result.push(positive);
          result.push(negative);
      }
      result
  }
}
